ddl/partition.go

```diff
@@ -759,6 +796,47 @@ func checkCreatePartitionValue(ctx sessionctx.Context, tblInfo *model.TableInfo)
 	return nil
 }
 
+func checkListPartitionValue(ctx sessionctx.Context, tblInfo *model.TableInfo) error {
+	pi := tblInfo.Partition
+	if len(pi.Definitions) == 0 {
+		return ast.ErrPartitionsMustBeDefined.GenWithStackByArgs("LIST")
+	}
+	if err := formatListPartitionValue(ctx, tblInfo); err != nil {
+		return err
+	}
+
+	var partitionsValuesMap []map[string]struct{}
+	partitionsValuesMap = append(partitionsValuesMap, make(map[string]struct{}))
+	for i := 1; i < len(pi.Columns); i++ {
```

Comment:
Since line 809 already append a map.

When `len(pi.Columns)` == 0, the `len(partitionsValuesMap)` should be 1, otherwise, the `len(partitionsValuesMap)` should be equal to `len(pi.Columns)`.